contracts/core/OptionsPremiumPricer.sol

```diff
@@ -0,0 +1,227 @@
+     * References for Black-Scholes:
+       https://www.macroption.com/black-scholes-formula/
+       https://www.investopedia.com/terms/b/blackscholes.asp
+       https://www.erieri.com/blackscholes
+       https://goodcalculators.com/black-scholes-calculator/
+       https://www.calkoo.com/en/black-scholes-option-pricing-model
+     * @param st is the strike price of the option
+     * @param expiryTimestamp is the unix timestamp of expiry
+     * @param isPut is whether the option is a put option
+     * @return premium for 100 contracts with 8 decimals i.e. 500*10**8 = $500 for 100 contracts
+     */
+    function getPremium(
+        uint256 st,
+        uint256 expiryTimestamp,
+        bool isPut
+    ) external view returns (uint256 premium) {
+        require(
+            expiryTimestamp > block.timestamp,
+            "Expiry must be in the future!"
+        );
+
+        (uint256 sp, uint256 v, uint256 t) =
+            blackScholesParams(expiryTimestamp);
+
+        (uint256 call, uint256 put) = quoteAll(t, v, sp, st);
+
+        premium = !isPut ? call : put;
+    }
+
+    /**
+     * @notice Calculates the option's delta
+     * Formula reference: `d_1` in https://www.investopedia.com/terms/b/blackscholes.asp
+     * http://www.optiontradingpedia.com/options_delta.htm
+     * https://www.macroption.com/black-scholes-formula/
+     * @param st is the strike price of the option
+     * @param expiryTimestamp is the unix timestamp of expiry
+     * @return delta for given option. 2 decimals (ex: 81 = 0.81 delta) as this is what strike selection
+     * module recognizes
+     */
+    function getOptionDelta(uint256 st, uint256 expiryTimestamp)
+        external
+        view
+        returns (uint256 delta)
+    {
+        require(
+            expiryTimestamp > block.timestamp,
+            "Expiry must be in the future!"
+        );
+
+        (uint256 sp, uint256 v, uint256 t) =
+            blackScholesParams(expiryTimestamp);
+
+        uint256 d1;
+        uint256 d2;
+
+        // Divide delta by 10 ** 12 to bring it to 2 decimals for strike selection
+        if (sp >= st) {
+            (d1, d2) = d(t, v, sp, st);
+            delta = Math.ncdf((Math.FIXED_1 * d1) / 1e18).div(10**12);
+        } else {
+            // If underlying < strike price notice we switch st <-> sp passed into d
+            (d1, d2) = d(t, v, st, sp);
+            delta = uint256(10)
+                .mul(10**13)
+                .sub(Math.ncdf((Math.FIXED_1 * d2) / 1e18))
+                .div(10**12);
+        }
+    }
+
+    /**
+     * @notice Calculates black scholes for both put and call
+     * @param t is the days until expiry
+     * @param v is the annualized volatility
+     * @param sp is the underlying price
+     * @param st is the strike price
+     * @return call is the premium of the call option given parameters
+     * @return put is the premium of the put option given parameters
+     */
+    function quoteAll(
```

Comment:
if you compare this to original keep3rvolatility [quoteAll](https://github.com/keep3r-network/keep3r.network/blob/a6897007db6e656e0e310ed7ee4ad42904fe2794/contracts/Keep3rV1Volatility.sol#L172), the only thing I changed was in st > sp condition, instead of _c = st-sp+_p we do _c = sp-st+_p.

their solution did not make sense because st > sp means that strike price is greater than underlying, which means the call should be less expensive than the put, which means that _c < _p so it should be (sp - st) which is negative as opposed to (st - sp) which is positive.